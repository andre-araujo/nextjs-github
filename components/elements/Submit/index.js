import React from 'react';
import { node } from 'prop-types';

import Button from './styles';

function Submit({
    children,
}) {
    return (
        <Button
            type="submit"
            value={children}
        />
    );
}

Submit.defaultProps = {
    children: 'Submit',
};

Submit.propTypes = {
    children: node,
};

export default Submit;
