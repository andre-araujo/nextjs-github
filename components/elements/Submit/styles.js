import styled from 'styled-components';

export default styled.input`
    background: transparent;
    border: 1px solid #333;
    color: #333;
    cursor: pointer;
    display: inline-block;
    font-family: AvantGardeDemi;
    font-size: 1.25rem;
    margin: 0;
    outline: none;
    padding: 0.625rem 2rem;
    transition: 0.2s background, 0.2s color;

    :hover {
        background: #333;
        color: #fafafa;
    }
`;
