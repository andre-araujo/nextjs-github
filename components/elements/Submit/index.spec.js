import React from 'react';
import { shallow } from 'enzyme';

import Submit from './index';

describe('Submit component', () => {
    it('should match snapshot with value as testing', () => {
        expect(shallow(
            <Submit>testing</Submit>,
        )).toMatchSnapshot();
    });

    it('should match snapshot with value as tests', () => {
        expect(shallow(
            <Submit>tests</Submit>,
        )).toMatchSnapshot();
    });
});
