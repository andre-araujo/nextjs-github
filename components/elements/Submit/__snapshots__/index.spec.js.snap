// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit component should match snapshot with value as testing 1`] = `
<styled.input
  type="submit"
  value="testing"
/>
`;

exports[`Submit component should match snapshot with value as tests 1`] = `
<styled.input
  type="submit"
  value="tests"
/>
`;
